woocommerce/woocommerce-gutenberg-products-block
# Convert TextInput and ValidatedTextInput to TypeScript

Before I get to work on allowing `TextInput` to be used as a `number` input (p1621348983029400-slack-C8X6Q7XQU) I converted the components to TypeScript.

### How to test the changes in this Pull Request:

1. Smoke test of the Cart and Checkout blocks.
2. Enter invalid information into checkout fields, ensure the appropriate validation error message is shown.
3. Ensure the error message goes away when you fix the error.
